src/tools/run-make-support/src/lib.rs

```diff
@@ -208,6 +209,42 @@ pub fn bin_name(name: &str) -> String {
     if is_windows() { format!("{name}.exe") } else { name.to_string() }
 }
 
+/// Remove all dynamic libraries possessing a name starting with `paths`.
+#[track_caller]
+pub fn remove_dylibs(paths: &str) {
+    let paths = format!(r"{paths}*");
+    remove_glob(dynamic_lib_name(&paths).as_str());
+}
```

Comment:
Discussion: these APIs looks a bit weird to me for a couple of reasons:

- It's based on globbing which can be surprising if you don't document the edge cases, and even then it's not immediately obvious how the implementation behaves.
- It uses `dynamic_lib_name` et al. on constructed glob patterns, which seems a bit weird to me, I think it's cleaner if we extract out `dynamic_lib_extension` that's used by both `dynamic_lib_name` and the fs walker to determine what files to remove.
- The count APIs are *especially* weird because I think it's almost an anti-pattern, for vast majority of tests I would expected that we control and know *exactly* which output artifacts we generate. Not checking for those (e.g. by removing the expected output artifacts) almost smell like bugs.

Maybe take a look at how tidy does filewalking (e.g. <https://github.com/rust-lang/rust/blob/master/src/tools/tidy/src/walk.rs>) for inspiration, but we definitely don't want to directly mimic tidy's interfaces. 

I think it's easier to understand if we implement it in terms of `walkdir` or some other variant of explicit filesystem walk, and document if it is recursive or not.

What do you think?